homeassistant/components/volvooncall/binary_sensor.py

```diff
@@ -18,22 +18,35 @@ async def async_setup_platform(
     """Set up the Volvo sensors."""
     if discovery_info is None:
         return
-    async_add_entities([VolvoSensor(hass.data[DATA_KEY], *discovery_info)])
+    async_add_entities([VolvoSensor(hass, *discovery_info)])
 
 
 class VolvoSensor(VolvoEntity, BinarySensorEntity):
     """Representation of a Volvo sensor."""
 
-    @property
-    def is_on(self):
-        """Return True if the binary sensor is on, but invert for the 'Door lock'."""
-        if self.instrument.attr == "is_locked":
-            return not self.instrument.is_on
-        return self.instrument.is_on
+    def __init__(
+        self, hass: HomeAssistant, vin, component, attribute, slug_attr, coordinator
+    ):
+        """Initialize the sensor."""
+        VolvoEntity.__init__(
+            self, hass, vin, component, attribute, slug_attr, coordinator
+        )
 
-    @property
-    def device_class(self):
-        """Return the class of this sensor, from DEVICE_CLASSES."""
         if self.instrument.device_class in DEVICE_CLASSES:
```

Comment:
For the future: We should start using EntityDescription to define the different sensor types and their device classes.

https://github.com/home-assistant/core/blob/10356b93795abd8c21539d1b25ebe5ce2e457c09/homeassistant/components/baf/binary_sensor.py#L39-L46